spring-integration-core/src/main/java/org/springframework/integration/dispatcher/UnicastingDispatcher.java

```diff
@@ -137,18 +137,24 @@ private boolean doDispatch(Message<?> message) {
 		if (!handlerIterator.hasNext()) {
 			throw new MessageDispatchingException(message, "Dispatcher has no subscribers");
 		}
-		List<RuntimeException> exceptions = new ArrayList<RuntimeException>();
+		List<RuntimeException> exceptions = new ArrayList<>();
```

Comment:
We should create this array lazily because it's only used in error situations. 99.9999% not used.